### PR TITLE
Add a `dscanner` configuration file and fix some of the warnings

### PR DESCRIPTION
**Pull request by Geod24**
A lot of lines, but not a lot of work. It did flag some interesting things though.

### CODE REVIEW

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/agora/node/FullNode.d [hunk 1]
```diff
@@ -739,17 +739,15 @@ public class FullNode : API
...
-            taskman = a TaskManager instance
+        May be overriden in unittests to allow test-adjusted clock times.
 
         Returns:
             an instance of a Clock
 
     ***************************************************************************/
 
-    protected Clock makeClock (ITaskManager taskman)
+    protected Clock makeClock ()
```
Comment (linked0) [inline source/agora/node/FullNode.d:750]:
```suggestion
    protected Clock makeClock (ITaskManager)
```

**Review by linked0 — APPROVED** [review]
LGTM, but one should be fixed.

**Comment by Geod24** [comment]
Should be fixed. Ran across a language limitation in the process...

**Pull request merged**